src/native/tab_bar.rs

```diff
@@ -380,68 +419,147 @@ where
-/// able to use a [`TabBar`](TabBar) in your user interface.
-pub trait Renderer: iced_native::Renderer {
-    /// The style supported by this renderer.
-    type Style: Default;
-
-    /// The default icon size of a [`TabBar`](TabBar).
-    const DEFAULT_ICON_SIZE: u16;
-
-    /// The default text size of a [`TabBar`](TabBar).
-    const DEFAULT_TEXT_SIZE: u16;
-
-    /// The default close size of a [`TabBar`](TabBar).
-    const DEFAULT_CLOSE_SIZE: u16;
-
-    /// The default padding of a [`TabBar`](TabBar).
-    const DEFAULT_PADDING: u16;
-
-    /// The default spacing of a [`TabBar`](TabBar).
-    const DEFAULT_SPACING: u16;
-
-    /// Draws a [`TabBar`](TabBar).
-    fn draw(
-        &mut self,
-        env: DrawEnvironment<'_, Self::Defaults, Self::Style, ()>,
-        active_tab: usize,
-        tab_labels: &[TabLabel],
-        icon_font: Option<Font>,
-        text_font: Option<Font>,
-    ) -> Self::Output;
-}
-
-#[cfg(debug_assertions)]
-impl Renderer for iced_native::renderer::Null {
-    type Style = ();
-
-    const DEFAULT_ICON_SIZE: u16 = 0;
-
-    const DEFAULT_TEXT_SIZE: u16 = 0;
-
-    const DEFAULT_CLOSE_SIZE: u16 = 0;
-
-    const DEFAULT_PADDING: u16 = 0;
-
-    const DEFAULT_SPACING: u16 = 0;
-
-    fn draw(
-        &mut self,
-        _env: DrawEnvironment<'_, Self::Defaults, Self::Style, ()>,
-        _active_tab: usize,
-        _tab_labels: &[TabLabel],
-        _icon_font: Option<Font>,
-        _text_font: Option<Font>,
-    ) -> Self::Output {
-    }
+#[allow(clippy::borrowed_box, clippy::too_many_lines)]
+fn draw_tab<Renderer>(
+    renderer: &mut Renderer,
+    tab: &TabLabel,
+    layout: Layout<'_>,
+    style_sheet: &Box<dyn StyleSheet>,
+    is_selected: bool,
+    cursor_position: iced_native::Point,
+    icon_font: Font,
+    text_font: Font,
+) where
+    Renderer: iced_native::Renderer + iced_native::text::Renderer,
+{
+    let is_mouse_over = layout.bounds().contains(cursor_position);
+    let style = if is_mouse_over {
+        style_sheet.hovered(is_selected)
+    } else {
+        style_sheet.active(is_selected)
+    };
+
+    let bounds = layout.bounds();
+    let mut children = layout.children();
+    let label_layout = children
+        .next()
+        .expect("Graphics: Layout should have a label layout");
+    let mut label_layout_children = label_layout.children();
+
+    renderer.fill_quad(
+        renderer::Quad {
+            bounds,
+            border_radius: 0.0,
+            border_width: style.tab_label_border_width,
+            border_color: style.tab_label_border_color,
+        },
+        style.tab_label_background,
+    );
+
+    match tab {
+        TabLabel::Icon(icon) => {
+            let icon_bounds = label_layout_children
+                .next()
+                .expect("Graphics: Layout should have an icon layout for an Icon")
+                .bounds();
+
+            renderer.fill_text(iced_native::text::Text {
+                content: icon.to_string(),
```

Comment:
The content of the new text drawing now accepts a string slice. Therefore I would like to avoid a heap allocation here. The only way I've found to turn a `char` into a `&str` is using the [`encode_uft8`](https://doc.rust-lang.org/std/primitive.char.html#method.encode_utf8) method as described [here](https://stackoverflow.com/a/47634755)

Something like:
```Rust
let mut buffer = [0; 4];
let icon = icon.encode_utf8(&mut buffer);

...
    content: icon,
...
```

It's not pretty but avoids allocating on the heap.